src/form/textarea/Textarea.tsx

```diff
@@ -0,0 +1,153 @@
+import "./_textarea.scss";
+
+import React, {useState} from "react";
+import classNames from "classnames";
+import TextareaAutosize, {TextareaAutosizeProps} from "react-textarea-autosize";
+
+import {KEYBOARD_EVENT_KEY} from "../../core/utils/keyboardEventConstants";
+
+type TextareaProps = Omit<
+  React.TextareaHTMLAttributes<HTMLTextAreaElement>,
+  "name" | "onChange" | "className"
+> & {
+  name: string;
+  onChange: React.ReactEventHandler<HTMLTextAreaElement>;
+  value?: string;
+  testid?: string;
+  isDisabled?: boolean;
+  customClassNames?: {
+    container?: string;
+    textarea?: string;
+  };
+  style?: React.CSSProperties;
+  autoSizeProps?: TextareaAutosizeProps;
+  onJustEnterPressed?: () => void;
+  onShiftEnter?: () => void;
+  isRequired?: boolean;
+  hasError?: boolean;
+};
+
+function Textarea(props: TextareaProps) {
+  const {
+    name,
+    value,
+    onChange,
+    placeholder,
+    isDisabled,
+    onFocus,
+    customClassNames,
+    onBlur,
+    testid,
+    onKeyUp,
+    id,
+    style,
+    autoSizeProps,
+    onJustEnterPressed,
+    isRequired,
+    onShiftEnter,
+    maxLength,
+    hasError,
+    autoComplete = "off",
+    autoCorrect = "off",
+    autoFocus = false,
+    ...otherProps
+  } = props;
+  const [isShiftPressed, setShiftPressedState] = useState(false);
+  const containerClassName = classNames(
+    "textarea-container",
+    customClassNames?.container
+  );
+  const textareaClassName = classNames("textarea", customClassNames?.textarea, {
+    "textarea--has-error": hasError
```

Comment:
Let's add this class to the container. Let's add a `is-disabled` modifier class to the container if the textarea is disabled as well